lib/Push.php

```diff
@@ -154,21 +205,38 @@ public function pushToDevice(int $id, INotification $notification, ?OutputInterf
 			return;
 		}
 
+		if ($this->deferPreparing) {
+			$this->notificationsToPush[$id] = clone $notification;
+			$this->loadDevicesForUsers[] = $notification->getUser();
+			$this->loadStatusForUsers[] = $notification->getUser();
```

Comment:
The device and status arrays can contain duplicated entries (for example, when starting a call in Talk there will be a voip notification and a background notification for the same user, which will cause the same user ID to be added twice to the arrays).

As far as I can tell that works fine, because the database queries for the [devices](https://github.com/nextcloud/notifications/pull/1161/files#diff-5f4fa2d752904832306c56f7832e2126a1e12db8dd1679abd71e526b77989821R607) and the [statuses](https://github.com/nextcloud/server/blob/5fcbb1ca62d7cfbffb0e4089f7315cdfe29668b0/apps/user_status/lib/Db/UserStatusMapper.php#L118) should return unique rows even if the ids were duplicated. But could that be a problem with some database **cough** oracle **cough**?

Reply:
Added array_unique in the flushing